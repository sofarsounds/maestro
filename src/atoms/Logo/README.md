# Logo

To implement a `Logo` into your project you'll need to add this import
```js
import { Logo } from '@sofarsounds/maestro'
```

After adding the import you can use any button simply like this
```html
<Logo />
```

## Props
Table below contains all types of props available in the Logo component  

| Name          | Type      | Default         | Description                      |
| :------------ | :-----    | :-------------- | :------------------------------- |
| circle        | `Boolean` | `false`         | Use `Circle` version
| cable         | `Boolean` | `false`         | Use `Cable` version
| invert        | `Boolean` | `false`         | Invert the Logo colour
| width         | `string ` | `false`         | Optional width for the logo
| className     | `string ` |                 | Optional className to add additional styling
| data-qaid     | `string`  |                 | Optional prop for testing purposes


