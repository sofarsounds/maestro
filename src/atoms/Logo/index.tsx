import React from 'react';
import styled, { css } from 'styled-components';

interface Props {
  circle?: boolean;
  cable?: boolean;
  invert?: boolean;
  width?: string;
  className?: string;
  'data-qaid'?: string;
  id?: string;
}

const LogoImg = styled.img`
  ${({ width }) => css`
    width: ${width || '100%'};
    margin: 0;
  `}
`;

const generateFileName = (circle: boolean, cable: boolean, invert: boolean) => {
  let kind = 'logo';
  if (circle) {
    kind = 'circle';
  }
  if (cable) {
    kind = 'cable';
  }

  const color = invert ? 'white' : 'black';

  return `sofar-${kind}-${color}.svg`;
};

const Logo: React.SFC<Props> = ({
  circle = false,
  cable = false,
  invert = false,
  width,
  className,
  'data-qaid': qaId,
  id
}) => (
  <LogoImg
    id={id}
    data-qaid={qaId}
    className={className}
    alt="Sofar Logo"
    src={require(`../../assets/${generateFileName(circle, cable, invert)}`)}
    width={width}
  />
);

export default Logo;
