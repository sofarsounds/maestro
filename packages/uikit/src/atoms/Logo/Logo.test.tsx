import React from 'react';
import { shallow } from 'enzyme';

import Logo from './index';

describe('Logo', () => {
  describe('Text Logo', () => {
    it('renders correctly', () => {
      expect(shallow(<Logo />)).toMatchSnapshot();
    });

    it('renders correctly when inverted', () => {
      expect(shallow(<Logo invert />)).toMatchSnapshot();
    });

    it('renders correctly with a given width', () => {
      expect(shallow(<Logo width="500px" />)).toMatchSnapshot();
    });
  });

  describe('Cable Logo', () => {
    it('renders correctly', () => {
      expect(shallow(<Logo cable />)).toMatchSnapshot();
    });

    it('renders correctly when inverted', () => {
      expect(shallow(<Logo cable invert />)).toMatchSnapshot();
    });

    it('renders correctly with a given width', () => {
      expect(shallow(<Logo cable width="500px" />)).toMatchSnapshot();
    });
  });

  describe('Circle Logo', () => {
    it('renders correctly', () => {
      expect(shallow(<Logo circle />)).toMatchSnapshot();
    });

    it('renders correctly when inverted', () => {
      expect(shallow(<Logo circle invert />)).toMatchSnapshot();
    });

    it('renders correctly with a given width', () => {
      expect(shallow(<Logo circle width="500px" />)).toMatchSnapshot();
    });
  });
});
