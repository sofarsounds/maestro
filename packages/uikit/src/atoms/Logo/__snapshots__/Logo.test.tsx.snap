// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Cable Logo renders correctly 1`] = `
<styled.img
  alt="Sofar Logo"
  src={[Function]}
/>
`;

exports[`Logo Cable Logo renders correctly when inverted 1`] = `
<styled.img
  alt="Sofar Logo"
  src={[Function]}
/>
`;

exports[`Logo Cable Logo renders correctly with a given width 1`] = `
<styled.img
  alt="Sofar Logo"
  src={[Function]}
  width="500px"
/>
`;

exports[`Logo Circle Logo renders correctly 1`] = `
<styled.img
  alt="Sofar Logo"
  src={[Function]}
/>
`;

exports[`Logo Circle Logo renders correctly when inverted 1`] = `
<styled.img
  alt="Sofar Logo"
  src={[Function]}
/>
`;

exports[`Logo Circle Logo renders correctly with a given width 1`] = `
<styled.img
  alt="Sofar Logo"
  src={[Function]}
  width="500px"
/>
`;

exports[`Logo Text Logo renders correctly 1`] = `
<styled.img
  alt="Sofar Logo"
  src={[Function]}
/>
`;

exports[`Logo Text Logo renders correctly when inverted 1`] = `
<styled.img
  alt="Sofar Logo"
  src={[Function]}
/>
`;

exports[`Logo Text Logo renders correctly with a given width 1`] = `
<styled.img
  alt="Sofar Logo"
  src={[Function]}
  width="500px"
/>
`;
